### PR TITLE
Add authentication support via an external agent

### PR DESCRIPTION
**Pull request by hagau**
This allows delegating the signature generation to an external agent.

### CODE REVIEW

File: sshlib/src/main/java/com/trilead/ssh2/auth/SignatureProxy.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 Jonas Dippel, Michael Perk, Marc Totzke
+ */
+
+package com.trilead.ssh2.auth;
+
+import java.io.IOException;
+import java.security.PublicKey;
+
+import jdk.nashorn.internal.objects.annotations.Getter;
```
Comment (jplitza) [inline sshlib/src/main/java/com/trilead/ssh2/auth/SignatureProxy.java]: This import as well as the `@Getter` line below caused problems during compilation for me (and looks like it doesn't belong here - Nashorn is a Javascript engine, isn't it!?)
After removing them, everything worked fine.

**Comment by hagau** [comment]
Hm, I'm apparently unable to sign the CLA for some reason. I have tried multiple times with different browser, for some reason it worked for [#35](https://github.com/connectbot/sshlib/pull/35), but not here.

File: sshlib/src/main/java/com/trilead/ssh2/auth/SignatureProxy.java (hunk 1 above)
Comment (hagau) [reply sshlib/src/main/java/com/trilead/ssh2/auth/SignatureProxy.java]: Thanks, I somehow overlooked that.
It's just an annotation and it should not cause problems with a recent enough JDK >= 8.
But looking at the definition of jdk.nashorn.internal.objects.annotations.Getter, it's not a general 'Getter' annotation but specific to a JS data property getter. I removed it.

**Comment by hagau** [comment]
@kruton Whoops, sorry for force pushing out your commits. I had just rebased :sweat_smile:

**Comment by hagau** [comment]
@kruton I have updated the patch to include support for RSA signatures with SHA512 & SHA256.

**Review by kruton — APPROVED** [review]

**Pull request merged**